Remove pathdiv from stubbs converters

ShellyHerself opened the pull request:
Moses sent me these as files :roll_eyes:

ShellyHerself committed:
Remove pathdiv from stubbs converters
(Moses)

csauve commented on tools_tag_converters/Convertor-Stubbs_Soso_to_Halo_OS_Soso.py:
```diff
@@ -62,7 +60,7 @@ def __init__(self, *args, **kwargs):
 
         self.tags_dir = StringVar(self)
-        self.tags_dir.set(curr_dir + 'tags' + PATHDIV)
+        self.tags_dir.set(curr_dir + 'tags/')
```
@MosesofEgypt Is this trailing `/` really necessary? Also shouldn't this be using `os.path.join(curr_dir, 'tags')`.

MosesofEgypt replied:
I'm pretty certain you can remove it from all the converters. I simply preserved the original code cause I was in a rush.

ShellyHerself commented on tools_tag_converters/Convertor-Stubbs_Coll_to_Halo_Coll.py:
```diff
@@ -71,8 +69,17 @@ def convert_coll_tag(coll_path):
     # move blocks from stubbs tag into the halo one
-    for i in (0, 1, 2, 3, 6, 7, 8):
-        tagdata[i] = stubbs_tagdata[i]
+    for attr_name in (
+            "flags",
+            "indirect_damage_material",
+            "body",
+            "shield",
+            "modifiers",
+            "pathfinding_box",
+            "pathfinding_spheres",
+            "nodes",
+            ):
+        tagdata[attr_name] = stubbs_tagdata[attr_name]
```
If you can type these out you might as well directly assign them.

MosesofEgypt replied:
i did test the stubbs converters before i sent them to you. i had to use them to help that one user in Reclaimers. That's how I noticed they crashed on compile.

ShellyHerself commented on tools_tag_converters/Convertor-Stubbs_Coll_to_Halo_Coll.py:
```diff
@@ -123,7 +130,7 @@ def __init__(self, *args, **kwargs):
 
         self.tags_dir = StringVar(self)
-        self.tags_dir.set(curr_dir + 'tags' + PATHDIV)
+        self.tags_dir.set(curr_dir + 'tags/')
```
What's the point of always appending the '/'? Is it so you can avoid using path.join()?

What happens if our starting path has backslashes?

MosesofEgypt replied:
As I stated above to csauve, the slash can be removed. This is a remnant of old habits dying hard.

ShellyHerself replied:
Could we handle that?

MosesofEgypt replied:
done. also, just tested all converters on windows and opened converted tags in mozz. tags load and appear fine.

ShellyHerself reviewed (commented):
There is an incredible amount of magic numbers in these files. And even though most of these were from before this PR. I am not comfortable merging it in this state.

This also hasn't been tested afaik. Not on linux at least. And I generally have low trust of files full of manual string operations on paths.

MosesofEgypt committed:
Replace for loops that copy fields with explicit statements

Explicitly copy animation fields rather than hackily changing descriptor
Only copy animation sound references, not animation effect references

MosesofEgypt committed:
Missed a couple fields in animation block copy

MosesofEgypt committed:
Clean up trailing forward slash in paths

ShellyHerself commented:
I'll be testing this later. I will be looking to get some test data as I don't have stubbz stuff on hand right now.

These changes are good so far

ShellyHerself commented:
In the mean time, maybe something to look into. These converters have version number strings embedded in their window titles. Those obviously wouldn't be correct anymore now. Maybe define those at the top of the files in a constant like we have for the mek_installer right now.

MosesofEgypt committed:
Move version string out of window title and make into a constant

ShellyHerself commented:
Testing these

ShellyHerself reviewed (commented):
This worked on my end. I would like the file dialogs to be switched over to the ones in binilla some time. But for now this is good enough

ShellyHerself commented:
Thank you for your effort, I appreciate that you changed so much
